### DISCUSSION TITLE
How to build a version that supports python 3.6

### DISCUSSION THREAD

**Discussion by javiqm12** [RUrlus/diptest#57] [Q&A] [answered]
Hi, first, thanks for this great work.

I need to install this dependency in an environment that only runs python 3.6
Can you give some hints on how to build a version of this package that supports python 3.6

thanks!

**Comment by RUrlus** [comment, marked as answer]
Hi @javiqm12,

That's not trivial but the main steps are:

1. Find latest PyBind11 version that supports 3.6 (I think 2.10 but not sure)
2. Find latest NumPy version that supports 3.6
3. Fix/Remove all type annotations
4. Correct minimum python version in pyproject.toml
5. Work through the other issues you encounter...

You shouldn't really face any issue on the C++ side of things (I think).

Given that Python 3.6 is well beyond end-of-life I dropped support a while ago now.
Good luck!

**Reply by javiqm12** [reply]
Thanks you very much for your answer. Yes, still using python 3.6 make troubles everywhere, hope admins will upgrade soon to a much newer version.
If I succeed, I will share the result here.
Have a nice day!
